### PR TITLE
Add Automatic PR for Updating MOOSE Submodule with Dependabot

### PR DESCRIPTION
**Pull request by nglaser3**
## Summary of changes

This PR adds a `dependabot.yml` file. Dependabot will automatically create a PR every Monday at 1:45 AM UTC updating the MOOSE submodule to it's most recent version. Dependabot will only create a PR when there are changes to the submodule. This config file is set-up so that Dependabot checks for updates once a week, and will only create a PR if there is no current PR open by dependabot. Further, the PR title will have a prefix of **_auto_update_moose_**, and will assign @smpark7 and myself to review, aswell as automatically assign `Comp:Core` and `Priority:1-Critical` labels.

Dependabot will not test whether the MOOSE update passes tests or not, but rather will rely on CIVET to run on the PR. 

This PR should not be merged until #281 has been addressed and resolved. 

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Required for Merging
- [x] I have read the **CONTRIBUTING** document.
- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.
  - [ ] CI tests pass
  - [ ] Local tests pass (including Serpent2 integration tests)

## Associated Issues and PRs

- Issue: #282 
- Issue: #281 

## Associated Developers

- Dev: @smpark7 

## Checklist for Reviewers

Reviewers should use [this link](https://arfc.github.io/manual/guides/pull_requests) to get to the 
Review Checklist before they begin their review.

### CODE REVIEW

File: .github/dependabot.yml [hunk 1]
```diff
@@ -0,0 +1,18 @@
+version: 2
+updates:
+  - package-ecosystem: "gitsubmodule"
+    directory: "/"
+    allow:
+      - dependency-name: "moose"
+    schedule:
+      interval: "weekly"
+      time: "01:45"
```
Comment (smpark7) [inline .github/dependabot.yml:9]: So it defaults to Monday if the day is not specified?

File: .github/dependabot.yml [hunk 2]
```diff
@@ -0,0 +1,18 @@
+version: 2
+updates:
+  - package-ecosystem: "gitsubmodule"
+    directory: "/"
+    allow:
+      - dependency-name: "moose"
```
Comment (smpark7) [inline .github/dependabot.yml:6]: Which moose branch does it pull from? next or devel or master? Their "main" branch is next, but I think devel would be safer from potential bugs

File: .github/dependabot.yml [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+      time: "01:45"
+    commit-message:
+      prefix: "auto_update_moose"
+    open-pull-requests-limit: 1 
+    labels:
+      - "Comp:Core"
+      - "Priority:1-Critical" 
+    reviewers:
+      - "smpark7"
+      - "nglaser3"
```
Comment (smpark7) [inline .github/dependabot.yml:18]: What does this do exactly? Does it tag us to review the PR?

**Review by smpark7 — COMMENTED** [review]
Thanks for setting this up! I have a few questions about the bot.

File: .github/dependabot.yml (hunk 3 above)
Comment (smpark7) [reply .github/dependabot.yml:18]: Okay that's what it does. I missed your description in the original post

File: .github/dependabot.yml (hunk 2 above)
Comment (nglaser3) [reply .github/dependabot.yml:6]: It pulls from the default branch, I am now realizing the default is the "next" branch. I will go ahead and try to fix this!

File: .github/dependabot.yml (hunk 1 above)
Comment (nglaser3) [reply .github/dependabot.yml:9]: Yes, the default is monday

File: .github/dependabot.yml (hunk 2 above)
Comment (nglaser3) [reply .github/dependabot.yml:6]: I have changed the target branch for the MOOSE submodule to the devel branch.

**Review by smpark7 — APPROVED** [review]
Alright LGTM. Last check before I merge? @nglaser3

**Comment by smpark7** [comment]
#281 has been addressed. Merging this PR

**Pull request merged**